src/core/server/saved_objects/routes/test_utils.ts

```diff
@@ -17,35 +17,29 @@
  * under the License.
  */
 
-import Hapi from 'hapi';
-// eslint-disable-next-line @kbn/eslint/no-restricted-paths
-import { defaultValidationErrorHandler } from '../../../../core/server/http/http_tools';
+import { ContextService } from '../../context';
+import { createHttpServer, createCoreContext } from '../../http/test_utils';
+import { coreMock } from '../../mocks';
 
-const defaultConfig = {
-  'kibana.index': '.kibana',
-  'savedObjects.maxImportExportSize': 10000,
-  'savedObjects.maxImportPayloadBytes': 52428800,
-};
+const coreId = Symbol('core');
+
+export const setupServer = async () => {
+  const coreContext = createCoreContext({ coreId });
+  const contextService = new ContextService(coreContext);
 
-export function createMockServer(config: { [key: string]: any } = defaultConfig) {
-  const server = new Hapi.Server({
-    port: 0,
-    routes: {
-      validate: {
-        failAction: defaultValidationErrorHandler,
-      },
-    },
+  const server = createHttpServer(coreContext);
+  const httpSetup = await server.setup({
+    context: contextService.setup({ pluginDependencies: new Map() }),
   });
-  server.config = () => {
-    return {
-      get(key: string) {
-        return config[key];
-      },
-      has(key: string) {
-        return config.hasOwnProperty(key);
-      },
-    };
-  };
+  const handlerContext = coreMock.createRequestHandlerContext();
 
-  return server;
-}
+  httpSetup.registerRouteHandlerContext(coreId, 'core', async (ctx, req, res) => {
```

Comment:
If you already created the mock, we can expose it and document a usage example. can be done in a separate PR ofc

Reply:
created https://github.com/elastic/kibana/issues/57844